### PR TITLE
Expose TransformManager.getParent(int)

### PR DESCRIPTION
**Pull request by ThomasGorisse**
It will be nice to have access to  TransformManager EntityInstance parent.

### CODE REVIEW

**Comment by ThomasGorisse** [comment]
@googlebot I signed it!

**Review by romainguy — APPROVED** [review]
Thanks!

File: android/filament-android/src/main/cpp/TransformManager.cpp [hunk 1]
```diff
@@ -86,6 +86,13 @@ Java_com_google_android_filament_TransformManager_nSetParent(JNIEnv*, jclass,
             (TransformManager::Instance) newParent);
 }
 
+extern "C" JNIEXPORT void JNICALL
```
Comment (romainguy) [inline android/filament-android/src/main/cpp/TransformManager.cpp:89]: You marked the function void but it returns a value. It should return jint

**Review by romainguy — CHANGES_REQUESTED** [review]

**Comment by ThomasGorisse** [comment]
Sorry for the last wrong include in the commit

**Comment by ThomasGorisse** [comment]
setParent takes an `Instance` parent parameter while getParent is returning an `Entiy`.
Is there a reason for the getParent return an Entity instead of an Instance [here](https://github.com/google/filament/blob/6b529bb2311e9a55d5c3293cc2b7b1dab738a25f/filament/src/components/TransformManager.cpp#L79) ?

**Comment by romainguy** [comment]
Returning an instance would be less useful. The entity gives you access to other properties (the Renderable etc.) and you can retrieve the instance from the Entity easily.

**Comment by ThomasGorisse** [comment]
Sorry for that failed pull request.
I have to learn again pointers before trying to PR on Filament.
My french engineering school might not be suitable for working in cpp environment ;-)

File: android/filament-android/src/main/cpp/TransformManager.cpp [hunk 2]
```diff
@@ -86,6 +86,13 @@ Java_com_google_android_filament_TransformManager_nSetParent(JNIEnv*, jclass,
             (TransformManager::Instance) newParent);
 }
 
+extern "C" JNIEXPORT jint JNICALL
+Java_com_google_android_filament_TransformManager_nGetParent(JNIEnv*, jclass,
+        jlong nativeTransformManager, jint i) {
+    TransformManager* tm = (TransformManager*) nativeTransformManager;
+    return tm->getParent((TransformManager::Instance) i);
```
Comment (romainguy) [inline android/filament-android/src/main/cpp/TransformManager.cpp:93]: You must cast to jint here: `return (jint) tm->getParent(…)`

**Review by romainguy — CHANGES_REQUESTED** [review]

**Comment by romainguy** [comment]
> Sorry for that failed pull request.
> I have to learn again pointers before trying to PR on Filament.
> My french engineering school might not be suitable for working in cpp environment ;-)

Pas de souci, je connais bien ce problème :)

**Comment by ThomasGorisse** [comment]
Should I delete this PR, rebase my fork and create a new PR with only the last acceptable check passed commit ?

**Comment by romainguy** [comment]
No it's fine don't worry about it. We always squash commits when merging PRs.

**Review by romainguy — APPROVED** [review]

**Comment by romainguy** [comment]
Thanks for the contribution!

**Pull request merged**